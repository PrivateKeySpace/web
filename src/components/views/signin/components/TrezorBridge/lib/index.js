export { default as stateSpec } from './stateSpec'
export { default as handlers } from './handlers'
