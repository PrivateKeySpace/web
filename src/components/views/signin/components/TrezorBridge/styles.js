import { StyleSheet } from 'aphrodite/no-important'
import { colors } from '../../../../../theme/constants'

const styles = StyleSheet.create({
  trezorBridge: {
    backgroundColor: colors.danger,
    color: colors.white
  }
})

export default styles
