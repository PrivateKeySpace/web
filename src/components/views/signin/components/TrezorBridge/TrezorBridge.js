import React from 'react'
import { compose, setDisplayName, pure, withState, withHandlers } from 'recompose'
import { css } from 'aphrodite/no-important'
import Select from '../../../../elements/Select'
import tx from '../../../../../theme/styles'
import sx from './styles'
import { stateSpec, handlers } from './lib'
import { DownloadButton } from './components'

const displayName = 'components/views/signin/components/TrezorBridge'

const enhance = compose(
  setDisplayName(displayName),
  ...stateSpec.map(statePropDefinition => withState(...statePropDefinition)),
  withHandlers(handlers),
  pure
)

function TrezorBridge (props) {
  const { hidden, selectedOS, handleSelectOS } = props

  return (
    <div className={css(tx.pt4, tx.pb3, tx.textAlignCenter, hidden && tx.hidden, sx.trezorBridge)}>
      <div className={css(tx.fontSize1, tx.textTransformUppercase)}>Trezor Bridge not installed</div>
      <div className={css(tx.fontSize3, tx.textTransformUppercase, tx.mb3)}>New communication tool to facilitate the connection between your TREZOR and your internet browser.</div>
      <div>
        <Select onChange={handleSelectOS} value={selectedOS}>
          <option value='windows'>Windows</option>
          <option value='macos'>macOS</option>
          <option value='linux'>Linux</option>
        </Select>
        <DownloadButton />
      </div>
    </div>
  )
}

export default enhance(TrezorBridge)
