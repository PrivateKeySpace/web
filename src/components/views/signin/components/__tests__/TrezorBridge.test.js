import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import TrezorBridge from '../TrezorBridge'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders without crashing and matches snapshot', () => {
  const layout = render(
    <TrezorBridge />
  )
  expect(layout).toMatchSnapshot()
})
