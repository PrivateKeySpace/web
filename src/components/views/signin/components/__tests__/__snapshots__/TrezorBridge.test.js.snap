// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<div
  class="pt4_34s7d-o_O-pb3_1a70m23-o_O-textAlignCenter_12to336-o_O-trezorBridge_129dtqa"
>
  <div
    class="fontSize1_3kvou9-o_O-textTransformUppercase_4efw5a"
  >
    Trezor Bridge not installed
  </div>
  <div
    class="fontSize3_o9hx2m-o_O-textTransformUppercase_4efw5a-o_O-mb3_1m9jzd0"
  >
    New communication tool to facilitate the connection between your TREZOR and your internet browser.
  </div>
  <div>
    <div
      class="p0_3js4c0-o_O-displayInlineBlock_36rlri-o_O-selectWrap_1hxi5gz"
    >
      <select
        class="p1_dxtwdx-o_O-select_6gixgy"
      >
        <option
          selected=""
          value="windows"
        >
          Windows
        </option>
        <option
          value="macos"
        >
          macOS
        </option>
        <option
          value="linux"
        >
          Linux
        </option>
      </select>
      <div
        class="arrow_vhhicj"
      >
        <span
          class="lnr lnr-chevron-down"
        >
          
        </span>
      </div>
    </div>
    <button
      class="p0_3js4c0-o_O-ml1_1dn7v29-o_O-textTransformUppercase_4efw5a-o_O-button_14du48h"
    >
      Download
    </button>
  </div>
</div>
`;
